src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -49,36 +55,47 @@ function MovieListPage() {
   }, [page, sortBy, sortOrder, searchQuery, selectedGenre]);
 
   const resetSearch = () => {
+    setTotalAmount(0);
     setMovies([]);
     setPage(1);
   }
 
   const handleSearch = (query: string) => {
```

Comment:
With the addition of the router, you can move all the filters and sorter to a separate component, e.g. "MovieFilters" and set query params there. Since `useSearchParams` acts like `useContext`, it should react to any URL changes and invoke the fetch. You'll have to move `resetSearch` somewhere, but there is only 1 place where it fits.
`handleSearch` in this case may be moved to either `Header` or `SearchForm`.

Reply:
Moved filter updates to the SearchForm and to the new MovieFilter component